documentation/http_api_index.markdown

```diff
@@ -0,0 +1,141 @@
+-   [Report](https://puppet.com/docs/puppet/latest/http_api/http_report.html)
+
+### Informational services
+
+These services are not directly used by Puppet agent, but can be used by other tools.
+
+-   [Status](https://puppet.com/docs/puppet/latest/http_status.html)
+
+### Environment endpoints
+
+The `/puppet/v3/environments` and `/puppet/v3/environment/:environment` endpoints use a different format than the configuration management and informational services endpoints.
+
+These endpoints accept only payloads formatted as JSON, and respond with JSON (MIME type of `application/json`).
+
+-   [Environments](https://puppet.com/docs/puppet/latest/http_api/http_environments.html)
+-   [Environment catalog](https://puppet.com/docs/puppet/latest/http_api/http_environment.html)
+
+### Puppet Server-specific endpoints
+
+Puppet Server adds several unique endpoints of its own. They include these additional `/puppet/v3/` endpoints:
+
+-   [Environment classes](./puppet-api/v3/environment_classes.markdown), at `/puppet/v3/environment_classes`
+-   [Environment modules](./puppet-api/v3/environment_modules.markdown), at `/puppet/v3/environment_modules`
+-   [Static file content](./puppet-api/v3/static_file_content.markdown), at `/puppet/v3/static_file_content`
+
+It also includes these unique APIs, with endpoints containing other URL prefixes:
+
+-   [Status API](./status-api/v1/services.markdown), at `/status/v1/services`
+-   [Metrics v1 (mbeans) API](./metrics-api/v1/metrics_api.markdown), at `/metrics/v1/mbeans`
+-   [Metrics v2 (Jolokia) API](./metrics-api/v2/metrics_api.markdown), at `/metrics/v2/`
+-   Admin API, at `/puppet-admin-api/v1/`:
+    -   [Environment cache](./admin-api/v1/environment-cache.markdown), at `/puppet-admin-api/v1/environment-cache`
+    -   [JRuby pool](./admin-api/v1/jruby-pool.markdown), at `/puppet-admin-api/v1/jruby-pool`
+
+### Error responses
+
+The `environments` endpoint responds to error conditions in a uniform manner and uses standard HTTP response codes to signify those errors.
+
+| Request problem | HTTP API error response code |
+|-----------------|------------------------------|
+| Client submits malformed request | 400 Bad Request |
+| Unauthorized client | 403 Not Authorized |
+| Client uses an HTTP method not permitted for the endpoint | 405 Method Not Allowed |
+| Client requests a response in a format other than JSON | 406 Unacceptable |
+| Server encounters an unexpected error while handling a request | 500 Server Error |
+| Server can't find an endpoint handler for an HTTP request | 404 Not Found |
+
+Except for HEAD requests, error responses contain a body of a uniform JSON object with the following properties:
+
+-   `message`: (`String`) A human-readable message explaining the error.
+-   `issue_kind`: (`String`) A unique label to identify the error class.
+
+There is also a [JSON schema for error objects](../schemas/error.json).
+
+## CA V1 HTTP API
```

Comment:
When running Puppet Server we use clojure based CA that does not go through the Ruby indirector based CA. While we worked hard to maintain compatibility with the Ruby indirector based CA and all this documentation should be correct as far as how to use the API is concerned, it does not, actually, go through the indirector any more.